docs/DeveloperGuide.md

```diff
@@ -255,66 +255,93 @@ _{Explain here how the data archiving feature will be implemented}_
+| `* * *` | year 4 SoC student with many contacts | search for contact of my TA/Prof | contact them when necessary |
+| `* *` | year 4 SoC student with many contacts | sort the contacts of my TA | view the contacts based on the sorting settings |
+| `* *` | careless student | undo my last action(s) | recover contacts I accidentally deleted/changed |
+| `* *` | organised SoC students | categorize the contacts of students/TA/Prof | view them separately |
+| `* *` | user with many contacts | add the profile picture of TA/Prof | more easily differentiate my contacts and remember their faces |
+| `*` | first time user | import existing contacts from my phone | easily access all my past contacts using SoConnect |
+| `*` | long term user | archive some contacts | still save old contacts without cluttering my current screen |
+
+#### Events
+
+| Priority | As a …​                                    | I want to …​                     | So that I can…​                                             |
+| -------- | ------------------------------------------ | ------------------------------ | ---------------------------------------------------------------------- |
+| `* * *` | SoC student | add details of event | keep track of the CCA dates and time |
+| `* * *` | SoC student | delete details of event | remove events that have finished or are cancelled |
+| `* * *` | SoC student | edit details of event | edit events that need to be edited |
+| `* * *` | SoC student | view all CCA/events | have a rough overview of my schedule | 
+| `* * *` | SoC student | search for an event based on event name | know more about the event |
+| `* *` | SoC student | sort the events by time | know which event is coming up next |
+| `* *` | SoC student with busy schedule | make sure my new event does not clash with any of the current event | better plan my timetable for additional activities |
+| `* *` | SoC student with many different events to manage | categorize my events according to different tags like classes and CCAs | easily filter and search for the events I want |
+| `*` | SoC student who uses other calendars | import and export my events to other calendars like Google Calendar | easily switch calendars to manage my events |
+| `*` | long term user | archive some events that have ended | still save details of past events without cluttering my current screen |
+| `*` | Soc student with many commitments | have a reminder for upcoming events | remember to attend those events |
+
+#### Personalisation and Others
+
+| Priority | As a …​                                    | I want to …​                     | So that I can…​                                             |
+| -------- | ------------------------------------------ | ------------------------------ | ---------------------------------------------------------------------- |
+| `* * *` | first time user | have a help message | know which are the basic features, which controls there are and how to start using |
```

Comment:
add "i can refer to" after help message
Change last column to "understand and start using basic features"